lib/topics_filter.rb

```diff
@@ -40,6 +40,22 @@ def filter_from_query_string(query_string)
         filter_created_by_user(usernames: values.flat_map { |value| value.split(",") })
       when "in"
         filter_in(values: values)
+      when "likes-min"
```

Comment:
there is a lot of repetition around `min = values.last` and `break if !integer_string?(min)`

I was thinking, if we can extract it to separate method and assign value on top of that switch.

Something like:

value = extract_and_validate_value_for("likes-min", values)

for category it would return `values` and for likes-max it would return `values.last`

In that case we would be able to extract constants like VALUE_MUST_BE_INTEGER = ["likes-max", "likes-op-min"...]. Later maybe we will have VALUE_MUST_BE_BOOLEAN or VALUE_MUST_BE_STRING

We will have some separation. One method would be responsible for transforming and validating data and another will be responsible for filtering:
```
filter_values = extract_and_validate_value_for(filter, values)
 case filter
      when "category"
        filter_categories(values: key_prefixes.zip(filter_values))
      when "created-by"
        filter_created_by_user(usernames: filter_values.flat_map { |value| value.split(",") })
      when "in"
        filter_in(values: filter_values)
        ...
        ```

How does it sound to you?

Reply:
Good idea :+1: I've introduce a method to remove the duplication here.